inference/models/sam2/segment_anything2.py

```diff
@@ -8,34 +8,38 @@
 
```

Comment:
is line `self.embedding_cache_keys.pop(0)` correct? I mean the 0 as key to pop

Reply:
`embed_image(...)` lack of typing

Reply:
why `score = score[sorted_ind]` is ignored?

Reply:
>is line self.embedding_cache_keys.pop(0) correct? I mean the 0 as key to pop
yeas I think we want FIFO for cache

Reply:
a ok, `embedding_cache_keys` is list not dict 👍